### PR TITLE
[papi] List editor options

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description

Implements the server's `getIDEOptions` in a new `EditorService` named `ListEditorOptions`. It is a stripped down version of the original method, only exposing these fields:
- `id`
- `title`
- `type`
- `logo`
- `label`
- `version`

Sample response:
```json
{
  "result": [
    {
      "id": "code",
      "title": "VS Code",
      "type": "TYPE_BROWSER",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/vscode.svg",
      "label": "Browser",
      "stable": {
        "version": "1.81.1"
      },
      "latest": {
        "version": "1.81.1"
      }
    },
    {
      "id": "code-desktop",
      "title": "VS Code",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/vscode.svg",
      "stable": {},
      "latest": {}
    },
    {
      "id": "intellij",
      "title": "IntelliJ IDEA",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/intellijIdeaLogo.svg",
      "label": "Ultimate",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2.1"
      }
    },
    {
      "id": "goland",
      "title": "GoLand",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/golandLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2"
      }
    },
    {
      "id": "pycharm",
      "title": "PyCharm",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/pycharmLogo.svg",
      "label": "Professional",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2.1"
      }
    },
    {
      "id": "phpstorm",
      "title": "PhpStorm",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/phpstormLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2.1"
      }
    },
    {
      "id": "rubymine",
      "title": "RubyMine",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/rubymineLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2.1"
      }
    },
    {
      "id": "webstorm",
      "title": "WebStorm",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/webstormLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2.1"
      }
    },
    {
      "id": "rider",
      "title": "Rider",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/riderLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2"
      }
    },
    {
      "id": "clion",
      "title": "CLion",
      "type": "TYPE_DESKTOP",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/clionLogo.svg",
      "stable": {
        "version": "2023.2"
      },
      "latest": {
        "version": "2023.2"
      }
    },
    {
      "id": "xterm",
      "title": "Terminal",
      "type": "TYPE_BROWSER",
      "logo": "https://ide.ft-list-editors.preview.gitpod-dev.com/image/ide-logo/terminal.svg",
      "label": "Insiders",
      "stable": {
        "version": "1.0.0"
      },
      "latest": {
        "version": "1.0.0"
      }
    }
  ]
}
```

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at f88fc14</samp>

This pull request adds a new editor service to the public API, which allows clients to query the available IDE options for the user. It also removes the obsolete workspaces service code from the public API TypeScript package. It involves changes to the gitpod service API, the public API server, the editor service protobuf definition, and the generated code for the editor service in Go and TypeScript.

</details>

## Related Issue(s)

Fixes EXP-454

## How to test

```bash
curl 'https://api.ft-list-editors.preview.gitpod-dev.com/gitpod.experimental.v1.EditorService/ListEditorOptions' \
  -H 'content-type: application/json' \
  -H 'Authorization: Bearer API_TOKEN' \
  --data '{}' | jq .
```

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-list-editors</li>
	<li><b>🔗 URL</b> - <a href="https://ft-list-editors.preview.gitpod-dev.com/workspaces" target="_blank">ft-list-editors.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-list-editors-gha.15505</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [x] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/public-api-server/pkg/apiv1/editor_service.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+var _ v1connect.EditorServiceHandler = (*EditorService)(nil)
+
+type EditorService struct {
+	connectionPool proxy.ServerConnectionPool
+
+	v1connect.UnimplementedEditorServiceHandler
+}
+
+func (s *EditorService) ListEditorOptions(ctx context.Context, req *connect.Request[v1.ListEditorOptionsRequest]) (*connect.Response[v1.ListEditorOptionsResponse], error) {
```
Comment (akosyakov) [inline components/public-api-server/pkg/apiv1/editor_service.go:33]: do we need some happy path tests for it similar to other services?

**Review by akosyakov — APPROVED** [review]
generally looks good but i did not test

/hold 


if no-one objects when merge when you are ready.

**Comment by akosyakov** [comment]
`type` does not look nice if someone will use it like REST, not sure whether we can make it simple `browser` or `desktop`

File: components/public-api/gitpod/experimental/v1/editor_service.proto [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    message Version {
+        // The semantic version of the stable editor image.
+        string stable = 1;
+
+        // The semantic version of the latest editor image.
+        string latest = 2;
+    }
+
+    Version version = 6;
```
Comment (akosyakov) [inline components/public-api/gitpod/experimental/v1/editor_service.proto:47]: I won't do it like that we may have later other different properties for latest or stable. I would rather make it flat `option.latestVersion` or have `option.latest.version`

File: components/public-api-server/pkg/apiv1/editor_service.go (hunk 1 above)
Comment (filiptronicek) [reply components/public-api-server/pkg/apiv1/editor_service.go:33]: Good point about adding tests, will add them.

**Comment by filiptronicek** [comment]
@akosyakov I'm not sure how to make pretty types. Protobuf enums look strict:
```
gitpod/experimental/v1/editor_service.proto:26:7:Enum value name "unspecified" should be prefixed with "EDITOR_TYPE_".
gitpod/experimental/v1/editor_service.proto:26:7:Enum value name "unspecified" should be UPPER_SNAKE_CASE, such as "UNSPECIFIED".
gitpod/experimental/v1/editor_service.proto:26:7:Enum zero value name "unspecified" should be suffixed with "_UNSPECIFIED".
gitpod/experimental/v1/editor_service.proto:27:7:Enum value name "browser" should be prefixed with "EDITOR_TYPE_".
gitpod/experimental/v1/editor_service.proto:27:7:Enum value name "browser" should be UPPER_SNAKE_CASE, such as "BROWSER".
gitpod/experimental/v1/editor_service.proto:28:7:Enum value name "desktop" should be prefixed with "EDITOR_TYPE_".
gitpod/experimental/v1/editor_service.proto:28:7:Enum value name "desktop" should be UPPER_SNAKE_CASE, such as "DESKTOP".
```

File: components/public-api/gitpod/experimental/v1/editor_service.proto (hunk 2 above)
Comment (filiptronicek) [reply components/public-api/gitpod/experimental/v1/editor_service.proto]: Made it the latter:
```
  "stable": {
    "version": "1.0.0"
  },
  "latest": {
    "version": "1.0.0"
  }
```

File: components/public-api-server/pkg/apiv1/editor_service.go (hunk 1 above)
Comment (filiptronicek) [reply components/public-api-server/pkg/apiv1/editor_service.go:33]: Donezies ✅

**Pull request merged**